src/gaggiuino.ino

```diff
@@ -800,7 +800,7 @@ static inline void sysHealthCheck(float pressureThreshold) {
 
   /* This *while* is here to prevent situations where the system failed to get a temp reading and temp reads as 0 or -7(cause of the offset)
   If we would use a non blocking function then the system would keep the SSR in HIGH mode which would most definitely cause boiler overheating */
-  while (currentState.temperature <= 0.0f || currentState.temperature == NAN || currentState.temperature >= 170.0f) {
+  while (currentState.sensorTemperature <= 0.0f || currentState.sensorTemperature == NAN || currentState.sensorTemperature >= 170.0f) {
```

Comment:
Better to check for the upper bound with the offset. In case when the sensor lowers values and the actual temperature is higher.

Or even for max(sensorTemperature, temperature)